core/src/backend/audio/decoders.rs

```diff
@@ -472,7 +472,7 @@ impl Read for SubstreamTagReader {
         buf[..len].copy_from_slice(&data[..len]);
 
         drop(data);
-        *chunk = chunk.to_start_and_end(chunk.start() + len, chunk.end());
+        *chunk = chunk.to_start_and_end(len, 0);
```

Comment:
I'm confused, how is this ever going to be not be empty?

Reply:
You're right, this throws out the rest of the current chunk. Unfortunately this isn't the cause of the audio glitching in Aperture Science :/